packages/shared/scope-hoisting/src/link.js

```diff
@@ -621,71 +786,135 @@ export function link({
 
         // If it's a $id$exports.name expression.
         let name = isIdentifier(property) ? property.name : property.value;
-        let {identifier} = resolveSymbol(asset, name);
+        let {identifier} = resolveSymbol(asset, name, bundle);
 
         if (identifier == null) {
           return;
         }
 
-        let {parent, parentPath} = path;
-        // If inside an expression, update the actual export binding as well
-        if (isAssignmentExpression(parent, {left: path.node})) {
-          if (isIdentifier(parent.right)) {
-            maybeReplaceIdentifier(
-              parentPath.get<NodePath<Identifier>>('right'),
-            );
+        let parent = ancestors[ancestors.length - 2];
+        // // If inside an expression, update the actual export binding as well
```

Comment:
Is that comment outdated?

Reply:
ah yeah, I had to move the assignment expression handling to a separate visitor